hedera-node/hedera-mono-service/src/main/java/com/hedera/services/state/merkle/MerkleAccountState.java

```diff
@@ -112,6 +113,11 @@ public class MerkleAccountState extends PartialMerkleLeaf implements MerkleLeaf
     // When staking to a node it is stored as -node-1 in order to differentiate nodeId=0
     private long stakedNum;
     private boolean declineReward;
+    // The entity expiration system task toggles on this flag when it reaches this account
+    // and finds it expired; we need this to prevent a zero-balance account with a funded
+    // auto-renew account from being treated as expired in the interval between its expiration
+    // and the time the system task actually auto-renews it
+    private boolean expiredAndPendingRemoval;
```

Comment:
This flag **is** `ON` only if the account is expired; and we will toggle it `ON` ASAP (as soon as the `ENTITY_EXPIRATION` system task gets a chance to process this account).

The problem is that without this flag, whenever `expiry < consensusTime` we have to do multiple checks on the status of the account's auto-renew account (if present), before we know if it's truly expired.

These checks are relatively expensive; and to avoid duplicating the logic that `ENTITY_EXPIRATION` already has, we'd need a larger refactor than in this fix.

Not sure if any of that helped. 🤔